File: services/payment-service-dotnet/src/PaymentService/Domain/Payment.cs
using System.ComponentModel.DataAnnotations;

namespace PaymentService.Domain
{
    public sealed class Payment
    {
        [Key]
        public Guid PaymentId { get; set; } = Guid.NewGuid();

        public Guid OrderId { get; set; }
        public Guid UserId { get; set; }

        public decimal Amount { get; set; }
        public string Currency { get; set; } = "DKK";

        public PaymentStatus Status { get; set; } = PaymentStatus.Pending;

        public string Provider { get; set; } = "MockProvider";
        public string? FailureReason { get; set; }

        public DateTimeOffset CreatedAt { get; set; } = DateTimeOffset.UtcNow;
    }
}

File: services/api-gateway-go/internal/http/dto/payment.go
package dto

import "time"

type Payment struct {
	PaymentID     string    `json:"paymentId"`
	OrderID       string    `json:"orderId"`
	UserID        string    `json:"userId"`
	Amount        float64   `json:"amount"`
	Currency      string    `json:"currency"`
	Status        int       `json:"status"`
	Provider      string    `json:"provider"`
	FailureReason *string   `json:"failureReason,omitempty"`
	CreatedAt     time.Time `json:"createdAt"`
}
